static/app/views/dashboards/widgetBuilder/components/visualize.tsx

```diff
@@ -0,0 +1,275 @@
+  );
+
+  // TODO: no parameters should show up as primary options?
+  // let aggregateOptions = Object.values(fieldOptions).filter(
+  //   option => option.value.kind === 'function' && option.value.meta.parameters.length > 0
+  // );
+  const aggregateOptions = useMemo(
+    () =>
+      Object.values(fieldOptions).filter(option =>
+        datasetConfig.filterYAxisOptions?.(state.displayType ?? DisplayType.TABLE)(option)
+      ),
+    [fieldOptions, state.displayType, datasetConfig]
+  );
+
+  // Used to extract selected aggregates and parameters from the fields
+  const stringFields = fields?.map(generateFieldAsString);
+
+  return (
+    <Fragment>
+      <SectionHeader
+        title={t('Visualize')}
+        tooltipText={t(
+          "This is the data you'd be visualizing in the display. If the overlay units conflict, the charts will always base it off of the first line."
+        )}
+      />
+      <Fields>
+        {fields?.map((field, index) => {
+          // Depending on the dataset and the display type, we use different options for
+          // displaying in the column select.
+          // For charts, we show aggregate parameter options for the y-axis as primary options.
+          // For tables, we show all string tags and fields as primary options, as well
+          // as aggregates that don't take parameters.
+          const columnOptions = Object.values(fieldOptions)
+            .filter(option => {
+              return (
+                // TODO: This should allow for aggregates without parameters
+                option.value.kind !== FieldValueKind.FUNCTION &&
+                (datasetConfig.filterYAxisAggregateParams?.(
+                  field,
+                  state.displayType ?? DisplayType.TABLE
+                )?.(option) ??
+                  true)
+              );
+            })
+            .map(option => ({
+              value: option.value.meta.name,
+              label:
+                state.dataset === WidgetType.SPANS
+                  ? prettifyTagKey(option.value.meta.name)
+                  : option.value.meta.name,
+            }));
+
+          return (
+            <FieldRow key={index}>
+              <FieldBar data-testid={'field-bar'}>
+                <ColumnCompactSelect
+                  searchable
+                  options={columnOptions}
+                  value={parseFunction(stringFields?.[index] ?? '')?.arguments[0] ?? ''}
+                  onChange={newField => {
+                    // TODO: Handle scalars (i.e. no aggregate, for tables)
+                    // Update the current field's aggregate with the new aggregate
+                    if (field.kind === 'function') {
+                      field.function[1] = newField.value as string;
+                    }
+                    dispatch({
+                      type: updateAction,
+                      payload: fields,
+                    });
+                  }}
```

Comment:
is there going to be a way to clear this field or select none after you've selected something? I noticed that there's aggregate functions that don't take any columns so i'm just wondering 👀

Reply:
Yeah I'm going to follow up on how these different fields work. The designs haven't really taken into account all of the "different" kind of aggregates, e.g. we have aggregates like `count_if` that take 3 args.

We could possibly allow clearing the column selector, that would be aligned with how a single field can be selected in the table (we're going to allow selecting `None` in the aggregates). I thought it'd be intuitive to make it a primary option in the column selector, and special case it so it doesn't show the aggregate selection where it doesn't make sense, but we'll see what works.

Reply:
ok ya sounds good! i'll slap the approve